cli/azd/extensions/microsoft.azd.extensions/internal/resources/languages/javascript/azdClient.js

```diff
@@ -0,0 +1,57 @@
+const grpc = require('@grpc/grpc-js');
+const protoLoader = require('@grpc/proto-loader');
+const path = require('path');
+
+class AzdClient {
+  constructor() {
+    const server = process.env.AZD_SERVER;
+    const token = process.env.AZD_ACCESS_TOKEN;
+
+    if (!server || !token) {
+      throw new Error('AZD_SERVER and AZD_ACCESS_TOKEN must be set.');
+    }
+
+    this._metadata = new grpc.Metadata();
+    this._metadata.add('authorization', token);
+
+    const options = {
+      keepCase: true,
+      longs: String,
+      enums: String,
+      defaults: true,
+      oneofs: true,
+    };
+
+    // Loads a .proto file and returns the generated gRPC package
+    const loadService = (name) => {
+      const protoPath = path.join(process.cwd(), 'proto', `${name}.proto`);
+      const packageDefinition = protoLoader.loadSync(protoPath, options);
+      const proto = grpc.loadPackageDefinition(packageDefinition);
+
+      return proto.azdext;
+    };
+
+    const address = server.replace(/^https?:\/\//, '');
+
+    // Load each proto service
+    const composeProto = loadService('compose');
+    const deploymentProto = loadService('deployment');
+    const environmentProto = loadService('environment');
+    const eventProto = loadService('event');
+    const projectProto = loadService('project');
+    const promptProto = loadService('prompt');
+    const userConfigProto = loadService('user_config');
+    const workflowProto = loadService('workflow');
```

Comment:
I was playing around with some of the issue here with they dynamic gRPC client generation from the proto files and the way assets are packaged with the `exe`.  

Are there other libraries that can be used that brings us closer to statically generated clients similar to the other languages? Maybe that would be a better/easier experience?

Reply:
We are currently revising this section, but it will take some time. If this PR is urgent, feel free to merge it first, and we will follow up with another PR to track the changes later.